src/main/scala/utility/ClockGatedReg.scala

```diff
@@ -25,4 +25,41 @@ object GatedValidRegNext {
     last := RegEnable(next, init, next || last)
     last
   }
+
+  def apply(last: Vec[Bool]): Vec[Bool] = {
+    val next = Wire(last.cloneType)
+    next := RegEnable(last, last.asUInt.orR || next.asUInt.orR)
```

Comment:
ok, I will optimize that.

Reply:
And the orR and `||` is for control signal clockgate, which need catch the signal when it is 1.

Reply:
It seems that the code here is controlling a Vec[bool] using one single enable signal (which is orR).
I suppose this is used because EDA will not insert gate cells when the number bits controlled by the enable signal is less than "min-width (usually 2-4)"?
This insight is not very intuitive to others. So I suggest adding comments about it.

Reply:
yes, I will add these comments.